### PR TITLE
LARA authors can mark an activity to use the activity player

### PR DESCRIPTION
**Pull request by emcelroy**
https://www.pivotaltracker.com/story/show/174824765

[#174824765]

* Add AP-only option to activity & sequence. 
* Limit embeddable types for AP-only activities/sequences
* Move AP url methods to models.

### CODE REVIEW

**Comment by emcelroy** [comment]
@scytacki Do you think it's necessary to correct the issues identified by Code Climate?

**Review by scytacki — CHANGES_REQUESTED** [review]
I found a few things to change.

File: app/views/lightweight_activities/new.html.haml [hunk 1]
```diff
@@ -29,11 +29,15 @@
       = f.select :project_id, options_from_collection_for_select(Project.all, 'id', 'title', @activity.project_id), { :include_blank => true }
     = field_set_tag "Official CC Activity" do
       = f.check_box :is_official
+      (Official CC activities appear in the STEM Resource Finder.)
```
[the remark this answers is not in the document]
Comment (emcelroy) [reply app/views/lightweight_activities/new.html.haml]: @scytacki Guess I assumed it corresponded with the similar option in portal settings, but it doesn't. Yeah, it doesn't seem to do much of anything in LARA either, though I'm not 100% sure. It's used to determine whether to add an "official" or "community" class to activity items listed on pages that list activities, but those classes don't seem to be used for anything.

I see some places where the option also seems to be used for activity filtering, but I'm not aware of any pages in LARA where filtering options like those are available.

I've removed the note I added, but left the option in for now.

File: app/models/publishable.rb [hunk 2]
```diff
@@ -39,7 +39,11 @@ def portal_publish(user,auth_portal,self_url)
   end
 
   def republish_for_portal(auth_portal,self_url,json=nil)
-    portal_publish_with_token(auth_portal.secret,auth_portal,self_url,true,json)
+    if self.runtime == "Activity Player"
+      portal_publish_with_token(user.authentication_token(auth_portal.strategy_name),auth_portal,self_url,true,json)
```
[the remark this answers is not in the document]
Comment (emcelroy) [reply app/models/publishable.rb:48]: @scytacki I'm afraid I've forgotten, too. Sorry. Does it maybe have to do with the fact that if the runtime is Activity Player, we use an API, but if the runtime is LARA, we don't?

image

**Review by scytacki — CHANGES_REQUESTED** [review]
The changes look good to me.
However this broke some spec tests, so you need to fix those before we can merge.

**Review by scytacki — APPROVED** [review]
Looks good to me.

**Pull request merged**